js/bundle.go

```diff
@@ -74,40 +78,40 @@ func newBundle(
 		return nil, err
 	}
 
-	// Compile sources, both ES5 and ES6 are supported.
-	code := string(src.Data)
-	c := compiler.New(piState.Logger)
-	c.Options = compiler.Options{
-		CompatibilityMode: compatMode,
-		Strict:            true,
-		SourceMapLoader:   generateSourceMapLoader(piState.Logger, filesystems),
-	}
-	pgm, _, err := c.Compile(code, src.URL.String(), false)
-	if err != nil {
-		return nil, err
-	}
 	// Make a bundle, instantiate it into a throwaway VM to populate caches.
-	rt := goja.New()
-	bundle := Bundle{
+	bundle := &Bundle{
 		Filename:          src.URL,
-		Source:            code,
-		Program:           pgm,
-		BaseInitContext:   NewInitContext(piState.Logger, rt, c, compatMode, filesystems, loader.Dir(src.URL)),
+		Source:            src.Data,
```

Comment:
Fixed in [aa208f8](https://github.com/grafana/k6/pull/2881/commits/aa208f8f92f3eedc0774ba0044c5074526bad510)

Looked at the same thing in lib.Archive ... but there were so many usages of it in hte test that I decided not to go and try to fix all of them :grimacing:

Reply:
Yeah, we can probably simplify a lot of things in that whole code path by using `TestPreInitState` and `TestRunState` more widely and expanding them... Or by grouping other similar "states" into their own structs instead of repeating a bunch of variables in multiple function argument lists and struct properties... :disappointed:

Definitely in separate PRs though, this PR already simplifies quite a few things :tada: